### PR TITLE
feat: reading count by timer and height

### PR DESCRIPTION
**Pull request by liruchen32**
# Issue
[[會員制]閱讀文章篇數-前端](https://app.asana.com/0/1203699264568808/1205988685100788/f)

# Notice

# Dependency

### CODE REVIEW

File: src/containers/Article.js [hunk 1]
```diff
@@ -85,6 +100,28 @@ class Article extends PureComponent {
           event: 'gtm.load',
         },
       })
+      this.startTimer()
+    }
+    if (this.state.isReachedTargetHeight && this.state.isReachedTargetTime) {
+      // TODO: waiting for api
+      // eslint-disable-next-line no-undef
+      alert(`你閱讀了 post_id: ${_.get(this.props.post, 'id')} 喔喔喔`)
```
Comment (Aylie-Chou) [inline src/containers/Article.js:108]: XD
alert `slug` may be easier to read & match (nice to have)

**Review by Aylie-Chou — CHANGES_REQUESTED** [review]
Mostly looking good 👍 
Please have a look at comments~~

File: src/containers/Article.js (hunk 1 above)
Comment (liruchen32) [reply src/containers/Article.js:108]: fixed

File: src/containers/Article.js [hunk 2]
```diff
@@ -50,16 +50,31 @@ class Article extends PureComponent {
...
+    }, 10000)
   }
 
   componentDidMount() {
     const { fontLevel, changeFontLevel, slugToFetch } = this.props
+    window.addEventListener('scroll', this.handleScroll)
 
     // Fetch the full post
     this.fetchAFullPostWithCatch(slugToFetch)
+    this.startTimer()
```
[the remark this answers is not in the document]
Comment (liruchen32) [reply src/containers/Article.js:77]: fixed

File: src/containers/Article.js [hunk 3]
```diff
@@ -50,16 +50,31 @@ class Article extends PureComponent {
     this.handleFontLevelChange = this._handleFontLevelChange.bind(this)
     this.onToggleTabExpanded = this._onToggleTabExpanded.bind(this)
     this._articleBody = React.createRef()
+    this.timerId = React.createRef()
     this.state = {
       isExpanded: false,
+      isReachedTargetHeight: false,
+      isReachedTargetTime: false,
     }
+    this.handleScroll = this.handleScroll.bind(this)
```
[the remark this answers is not in the document]
Comment (liruchen32) [reply src/containers/Article.js:59]: fixed

File: src/containers/Article.js [hunk 4]
```diff
@@ -85,6 +100,28 @@ class Article extends PureComponent {
...
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.handleScroll)
+  }
+
+  handleScroll() {
+    const scrollHeight = window.scrollY || document.documentElement.scrollTop
+    const totalHeight = document.documentElement.scrollHeight
+    const targetHeight = totalHeight * 0.75
```
[the remark this answers is not in the document]
Comment (liruchen32) [reply src/containers/Article.js:119]: fixed

**Review by Aylie-Chou — APPROVED** [review]
ＬＧＴＭ

**Pull request merged**